lib/blocs/search_locations/search_locations_cubit.dart

```diff
@@ -65,8 +65,13 @@ class SearchLocationsCubit extends Cubit<SearchLocationsState> {
   }
 
   void insertHistoryPlace(TrufiLocation location) {
-    emit(state.copyWith(historyPlaces: [...state.historyPlaces, location]));
-    historyPlacesStorage.insert(location);
+    emit(state.copyWith(historyPlaces: [
+      ..._deleteItem(state.historyPlaces, location),
+      location
+    ]));
+    historyPlacesStorage.delete(location).then(
+          (value) => historyPlacesStorage.insert(location),
+        );
```

Comment:
state.historyPlaces is already with the last state, what about if you just save it. then it can be saved in one operation instead two operations (delete and insert)